apps/admin_app/lib/admin_app_web/controllers/dashboard_controller.ex

```diff
@@ -0,0 +1,96 @@
+defmodule AdminAppWeb.DashboardController do
+  use AdminAppWeb, :controller
+
+  alias Snitch.Data.Model
+
+  def index(conn, params) do
+    {start_date, end_date} = get_date_from_params(params)
+
+    order_state_counts = get_order_state_count(start_date, end_date)
+    product_state_counts = get_product_state_count(start_date, end_date)
+    order_datepoints = get_order_datapoints(start_date, end_date)
+    payment_datapoints = get_payment_datapoints(start_date, end_date)
+
+    render(conn, "index.html",
+      order_state_counts: order_state_counts,
+      product_state_counts: product_state_counts,
+      order_datepoints: order_datepoints,
+      payment_datapoints: payment_datapoints
+    )
+  end
+
+  defp get_naive_date_time(date) do
+    Date.from_iso8601(date)
+    |> elem(1)
+    |> NaiveDateTime.new(~T[00:00:00])
+    |> elem(1)
+  end
+
+  defp get_date_from_params(params) do
+    today = Date.utc_today() |> Date.to_string() |> get_naive_date_time()
+
+    start_date =
+      case Map.get(params, "from") do
```

Comment:
This can be turned into a generic function you have used the same logic in `view` also.